app/models/manageiq/providers/openstack/cloud_manager/refresh_parser.rb

```diff
@@ -243,6 +249,25 @@ def parse_availability_zone(az, service_name)
       return uid, new_result
     end
 
+    def parse_host_aggregate(ha)
+      uid = ha.id
+      infra_ems = @ems.provider && @ems.provider.try(:infra_ems)
+      ems_hosts = infra_ems.try(:hosts)
+      hosts = ha.hosts.map do |fog_host|
+        ems_hosts.try(:find) { |h| h.hypervisor_hostname == fog_host.split('.').first }
+      end
```

Comment:
It looks to me that ```ha.hosts.map``` block can return array containg ```nil``` elements, I am not sure if it is ok for passing it to AR association.